### PR TITLE
docs: improve cookbook to share client session and modules with classes

### PR DESCRIPTION
**Pull request by TomChv**
Follow up to: #5721

### CODE REVIEW

**Comment by TomChv** [comment]
Result from my local build 🚀 

Screenshot 2023-09-21 at 15 03 47
Screenshot 2023-09-21 at 15 03 58
Screenshot 2023-09-21 at 15 04 07

File: docs/current/cookbook.md
Comment (vikram-dagger) [inline docs/current/cookbook.md:1014]:
```suggestion
Another possible approach is to use independent classes (or interfaces, depending on the programming language) with public methods as functions. The following code listing demonstrates this approach.
```

File: docs/current/cookbook.md
Comment (vikram-dagger) [inline docs/current/cookbook.md:1015]:
```suggestion
```

File: docs/current/cookbook.md
Comment (vikram-dagger) [reply docs/current/cookbook.md:1014]: Suggest changing the section header to "Organize pipeline code into reusable modules or classes"

**Review by vikram-dagger — COMMENTED** [review]
See comments. I will defer to @helderco @slumbering @kpenfound @gerhard  to review the code.

File: docs/current/cookbook/snippets/modules-shared-client/classes/pipelines/pipelines.go
Comment (gerhard) [inline docs/current/cookbook/snippets/modules-shared-client/classes/pipelines/pipelines.go:1]: Which one reads better to you?
- `pipelines.New()`
- `pipeline.New()`

**Review by gerhard — COMMENTED** [review]
This looks better! I've left a few improvement suggestions.

File: docs/current/cookbook/snippets/modules-shared-client/classes/pipelines/pipelines.go
Comment (TomChv) [reply docs/current/cookbook/snippets/modules-shared-client/classes/pipelines/pipelines.go:18]: should I change it in the other example too then?

File: docs/current/cookbook/snippets/modules-shared-client/classes/pipelines/pipelines.go
Comment (TomChv) [reply docs/current/cookbook/snippets/modules-shared-client/classes/pipelines/pipelines.go:10]: I agree!

**Review by vikram-dagger — APPROVED** [review]
LGTM to me, please incorporate other reviewer's comments before merging.

**Comment by TomChv** [comment]
@vikram-dagger @gerhard I updated the code according to your comment, let me know if I need to change anything else

**Comment by vikram-dagger** [comment]
This looks fine to me. I will wait for @gerhard to review and approve the code changes before merging.

**Review by gerhard — COMMENTED** [review]
LGTM 💪

**Pull request merged**
